faculty/clients/base.py

```diff
@@ -127,27 +120,23 @@ class BaseClient(object):
 
     SERVICE_NAME = None
 
-    def __init__(self, profile):
+    def __init__(self, session):
         if self.SERVICE_NAME is None:
             raise RuntimeError(
                 "must set SERVICE_NAME in subclasses of BaseClient"
             )
-        self.profile = profile
+        self.session = session
         self._http_session_cache = None
 
     @property
     def http_session(self):
```

Comment:
Now that we have a long-lived session object, I think it also makes sense to move the requests session there. This way the client base doesn't even need to know anything about authentication, it's all managed at the session level.

Reply:
Perhaps, since this was mostly an implementation detail of the `BaseClient`, I was planning on deferring this to a future PR.

There is a little complexity to be introduced by needing a slightly different `Auth` class for sending the token as a cookie instead of an HTTP header (for the hound client), and since that need to 'choose' an auth type is client-specific, I think it makes sense to wait until we have that before doing further refactoring.